XMonad/Prompt.hs

```diff
@@ -1803,7 +1810,8 @@ historyCompletionP p = do
     cd <- asks (cacheDir . directories)
     pure $ \x ->
         let toComplList = deleteConsecutive . filter (isInfixOf x) . M.foldr (++) []
-         in toComplList . M.filterWithKey (const . p) <$> readHistory cd
+--          in toComplList . M.filterWithKey (const . p) <$> (historyFetch conf) cd
+         in toComplList . M.filterWithKey (const . p) <$> (readHistory cd)
```

Comment:
I don't know if `historyCompletionP` can be updated without a breaking change, which I am indicating here by still using `readHistory`

Reply:
The problem, fwiw, is that `conf` isn't provided to `historyCompletionP`